iothub/device/src/InternalClient.cs

```diff
@@ -1303,28 +1253,33 @@ public Task<Message> ReceiveAsync(CancellationToken cancellationToken)
             }
         }
 
-        private Task DisableMethodAsync(CancellationToken cancellationToken)
+        private async Task HandleMethodDisableAsync(CancellationToken cancellationToken)
         {
-            return _deviceMethods == null && _deviceDefaultMethodCallback == null
-                ? InnerHandler.DisableMethodsAsync(cancellationToken)
-                : TaskHelpers.CompletedTask;
+            // Don't disable if it is already disabled or if there are registered device methods
+            if (!_isDeviceMethodEnabled
```

Comment:
Trying to be a little more careful about when to disable methods, so it still includes the default handler not being null or there being any specific methods.